### PR TITLE
Formal deprecation of 4 auth endpoint methods; rubocop

### PR DESCRIPTION
**Pull request by joshcanhelp**
### Changes

- Add deprecation notices on the following methods:
  - `Auth0::Api::AuthenticationEndpoints.obtain_access_token`
  - `Auth0::Api::AuthenticationEndpoints.obtain_user_tokens`
  - `Auth0::Api::AuthenticationEndpoints.login`
  - `Auth0::Api::AuthenticationEndpoints.user_info`
- Ran `rubocop` in `Auth0::Api::AuthenticationEndpoints` and fixed issues

### CODE REVIEW

File: lib/auth0/api/authentication_endpoints.rb [hunk 1]
```diff
@@ -75,45 +77,7 @@ def exchange_refresh_token(
         AccessToken.from_response post('/oauth/token', request_params)
       end
 
-      # Retrieve an access token.
```
Comment (joshcanhelp) [inline lib/auth0/api/authentication_endpoints.rb]: Moved down, added deprecation docs

File: lib/auth0/api/authentication_endpoints.rb [hunk 2]
```diff
@@ -75,45 +77,7 @@ def exchange_refresh_token(
...
-        if access_token
-          request_params = { client_id: @client_id, access_token: access_token, connection: connection, scope: scope }
-          post('/oauth/access_token', request_params)['access_token']
-        else
-          request_params = { client_id: @client_id, client_secret: @client_secret, grant_type: 'client_credentials' }
-          post('/oauth/token', request_params)['access_token']
-        end
-      end
-
-      # Get access and ID tokens using an Authorization Code.
```
Comment (joshcanhelp) [inline lib/auth0/api/authentication_endpoints.rb]: Moved down, added deprecation docs

File: lib/auth0/api/authentication_endpoints.rb [hunk 3]
```diff
@@ -136,49 +100,23 @@ def login_with_resource_owner(
...
-          grant_type:    realm ? 'http://auth0.com/oauth/grant-type/password-realm' : 'password'
+          realm: realm,
+          scope: scope,
+          audience: audience,
+          grant_type: realm ? 'http://auth0.com/oauth/grant-type/password-realm' : 'password'
         }
         AccessToken.from_response post('/oauth/token', request_params)
       end
-
-      # Get access and ID tokens using Resource Owner Password.
```
Comment (joshcanhelp) [inline lib/auth0/api/authentication_endpoints.rb]: Moved down, added deprecation docs

File: lib/auth0/api/authentication_endpoints.rb [hunk 4]
```diff
@@ -266,15 +208,7 @@ def wsfed_metadata
       # @see https://auth0.com/docs/api/authentication#get-user-info
       # @return [json] User information based on the Auth0 access token
       def userinfo(access_token)
-        get('/userinfo', {}, {'Authorization' => "Bearer #{access_token}"})
-      end
-
-      # Return the user information based on the Auth0 access token.
```
Comment (joshcanhelp) [inline lib/auth0/api/authentication_endpoints.rb]: Moved down, added deprecation docs

File: lib/auth0/api/authentication_endpoints.rb [hunk 5]
```diff
@@ -343,7 +278,7 @@ def wsfed_url(connection = UP_AUTH, options = {})
           wreply: options[:wreply]
         }
 
-        url_client_id = @client_id if !request_params[:wtrealm]
+        url_client_id = @client_id unless request_params[:wtrealm]
```
Comment (machuga) [inline lib/auth0/api/authentication_endpoints.rb]: 👏 👏 👏 👏 👏

**Pull request merged**